webui/react/src/components/Table/InteractiveTable.tsx

```diff
@@ -443,7 +446,7 @@ const InteractiveTable = <T extends object>({
 
       if (isEqual(newSettings, settings)) return;
 
-      updateSettings(newSettings);
+      updateSettings(newSettings as Partial<S>);
```

Comment:
not sure why but without this, it reports:
```
TS2345: Argument of type 'Partial ' is not assignable to parameter of type 'Partial'.
```

Reply:
`Partial<InteractiveTableSettings>` !== `Partial<S>`. If the caller uses a different value for `S` this code is incorrect.

Reply:
Are you sure you meant to make this component generic over the settings type?

Reply:
> `Partial<InteractiveTableSettings>` !== `Partial<S>`. If the caller uses a different value for `S` this code is incorrect.

Even if `S` is defined as `S extends InteractiveTableSettings`? Doesn't that ensure that whatever `S` type you provide needs to conform to `InteractiveTableSettings` at the minimum?

Reply:
> Are you sure you meant to make this component generic over the settings type?

uh... no? 😆 

lemme take a crack again with just using the `InteractiveTableSettings` type.

Reply:
`S` can have more fields than `InteractiveTableSettings` which the function could rely on which is why this is not valid as written (I know this is a special case due to `Partial` but typescript doesn't account for that).